newdle/client/src/util/date.js

```diff
@@ -2,28 +2,56 @@ import moment, {HTML5_FMT} from 'moment';
 
 export const DEFAULT_TIME_FORMAT = 'HH:mm';
 
-export function toMoment(date, format = null) {
-  return date ? moment(date, format) : null;
+export function toMoment(date, format = null, timezone = null) {
+  return date ? moment.tz(date, format, timezone) : null;
 }
 
-export function serializeDate(date, format = HTML5_FMT.DATE) {
-  return moment(date).format(format);
+export function serializeDate(date, format = HTML5_FMT.DATE, timezone = null) {
+  return moment.tz(date, timezone).format(format);
 }
 
 export function overlaps([start1, end1], [start2, end2]) {
   return start1.isBefore(end2) && start2.isBefore(end1);
 }
 
 export function getHourSpan(input) {
-  const {timeSlots, defaultHourSpan, defaultMinHour, defaultMaxHour, duration, format} = input;
-  const timeSlotsMoment = timeSlots.map(c => toMoment(c, format));
+  const {
+    timeSlots,
+    busyTimes,
```

Comment:
I belive busyTimes were not being considered for the answers page, so we wouldn't see them on the answers page even though react would try to renderize them. We should either: 
1 - Include them correctly on the display between a correct min and max hours (this approach)
2 - Don't show busy times that are outside min and max hours range

Reply:
I don't think we care much about busy times outside the displayed timespan here. So I'd just hide those.

Reply:
BTW, this indeed looks a bit weird (in master):

![image](https://user-images.githubusercontent.com/179599/91695326-3e289b80-eb6e-11ea-9530-8f6249d8b14a.png)

```
$ http -pb get 'https://newdle.blackhole.cern.ch/api/newdle/7kYbqYp9/participants/xprFxn2f/busy' 'date==2020-09-11' 'tz==Europe/Zurich'
[
    [
        "00:00",
        "00:00"
    ],
    [
        "14:00",
        "16:00"
    ]
]
```

However, I think this is actually a bug in the API. We should not return a zero-duration busy slot at all (this is coming from the all-day-busy on the day before).

Reply:
BTW, taking busy times into account here in any regard would probably be something for a separate issue/PR. Especially when it comes to timezone logic which is already tricky by itself, it does not make PRs easier to review and test if they include other, unrelated things as well. 

But nonetheless, thanks for realizing that this is a point which should probably be improved!

Reply:
Ok, I agree, I'll leave it out for now and create a separate issue

Reply:
Hey, I realized this discussion all steamed from me not understanding that busy slots below minHour will be assigned pos 0 and no height. I believed the minHour should be the minimum of the busy slots and options slots just because I didn't understand that I was rendering the busy slots wrongly. I now believe my timezone aware busySlots implementation is correct and they render correctly (pos and height wise), so I believe this discussion can be dropped

Reply:
Although I can open a new issue if we want to fix the red line that appears at the top when the busy slot is outside our display range

Reply:
i think they are fine - to indicate that there is *something* earlier, without cluttering the vertical space.

(i'd just fix the 00:00-00:00 entries sent e.g. by the exchange backend, because those are useless since they come from the previous day)

Reply:
I think I fixed this 00:00-00:00 shortcoming, please check it out